src/Microsoft.AspNet.Mvc.ModelBinding/Binders/MutableObjectModelBinder.cs

```diff
@@ -323,6 +323,12 @@ internal static PropertyValidationInfo GetPropertyValidationInfo(ModelBindingCon
             {
                 var propertyName = property.Name;
                 var propertyMetadata = bindingContext.ModelMetadata.Properties[propertyName];
+                if (propertyMetadata == null)
+                {
+                    // Skip indexer properties and others ModelMetadata ignores.
+                    continue;
+                }
```

Comment:
hit the NRE when I changed something else.  but I split it into a separate commit if we want to get it in separately.